### PR TITLE
Watch-only address balance should not be included in the overall balance by default #19657

### LINKED ISSUE #19657: Watch-only address balance should not be included in the overall balance by default
# Bug Report

## Problem 

The watch-only account balance is currently included in the overall balance display. Later, a setting will be added to change this, but for now, including watch-only balance in the overall balance display should be **disabled** by default, i.e. the balance on the main wallet screen should not include watch-only accounts balance.

Screenshot 2024-04-16 at 11 26 40

### Reproduction

1. Create a new profile
2. Add a W/O address

#### Expected result
The balance should be 0

### Additional Information

- Status version: nightly 16/04
- Operating System: Android, iOS

[comment]: # (Please, add logs/notes if necessary)

### PR DESCRIPTION
**Pull request by mmilad75**
fixes #19657 

https://github.com/status-im/status-mobile/assets/55688834/57ceed43-0ee9-4943-bb77-7a12e54e0cd6

### CODE REVIEW

**Review by ibrkhalil — APPROVED** [review]

**Comment by J-Son89** [comment]
One question, does this affect the balance of the individual tokens in the asset list too? 🤔 

I think we should check with designers about this.

Screenshot 2024-04-19 at 18 30 24

**Comment by mmilad75** [comment]
> One question, does this affect the balance of the individual tokens in the asset list too? 🤔
> 
> I think we should check with designers about this.
> 
> Screenshot 2024-04-19 at 18 30 24

No, it doesn't. The ticket was only for the top balance. I will check with them and if it's required, I will update them as well @J-Son89

**Comment by J-Son89** [comment]
> > One question, does this affect the balance of the individual tokens in the asset list too? 🤔
> > I think we should check with designers about this.
> > Screenshot 2024-04-19 at 18 30 24
> 
> No, it doesn't. The ticket was only for the top balance. I will check with them and if it's required, I will update them as well @J-Son89

cool, thanks @mmilad75 - we can keep them as separate issues either way. But yes let's check :)

**Comment by mmilad75** [comment]
> > > One question, does this affect the balance of the individual tokens in the asset list too? 🤔
> > > I think we should check with designers about this.
> > > Screenshot 2024-04-19 at 18 30 24
> > 
> > 
> > No, it doesn't. The ticket was only for the top balance. I will check with them and if it's required, I will update them as well @J-Son89
> 
> cool, thanks @mmilad75 - we can keep them as separate issues either way. But yes let's check :)

@J-Son89 It should be included https://discord.com/channels/624634427930312714/928625369542713396/1231915683294478368

**Comment by churik** [comment]
@mmilad75 @J-Son89 it is required; 
we do not have this setting and it is not implemented yet. 
The balances should match as a sum of assets and for the time being (unitl it this setting is implemented) should be not included in the total balance and the sum of assets.

**Comment by churik** [comment]
@mmilad75 would you mind to fix it here?

**Comment by mmilad75** [comment]
> @mmilad75 would you mind to fix it here?

Sure. I had issues with running the app. I'll be working on it today

File: src/status_im/config.cljs [hunk 1]
```diff
@@ -133,3 +133,5 @@
 (def community-accounts-selection-enabled? true)
 (def fetch-messages-enabled? (enabled? (get-config :FETCH_MESSAGES_ENABLED "1")))
 (def test-networks-enabled? (enabled? (get-config :TEST_NETWORKS_ENABLED "0")))
+
+(def include-watch-only-balance-in-total? false)
```
Comment (smohamedjavid) [inline src/status_im/config.cljs]: This flag can be removed as we are excluding the watched addresses in the balance calculation.

**Comment by mariia-skrypnyk** [comment]
Hi @mmilad75 !

Checked and still see sum of My + Watch Only tokens:
Screenshot 2024-04-25 at 19 51 08

File: src/status_im/subs/wallet/wallet.cljs [hunk 2]
```diff
@@ -291,8 +292,10 @@
 (rf/reg-sub
  :wallet/aggregated-tokens
  :<- [:wallet/accounts]
- (fn [accounts]
-   (utils/aggregate-tokens-for-all-accounts accounts)))
+ :<- [:wallet/accounts-without-watched-accounts]
+ (fn [[all-accounts owned-accounts]]
+   (let [accounts (if config/include-watch-only-balance-in-total? all-accounts owned-accounts)]
+     (utils/aggregate-tokens-for-all-accounts accounts))))
```
Comment (smohamedjavid) [inline src/status_im/subs/wallet/wallet.cljs:296]:
```clojure
(rf/reg-sub
 :wallet/aggregated-tokens
 :<- [:wallet/accounts-without-watched-accounts] ;; Replace :wallet/accounts with this sub
 (fn [accounts]
   (utils/aggregate-tokens-for-all-accounts accounts)))

```

File: src/status_im/config.cljs (hunk 1 above)
Comment (mmilad75) [reply src/status_im/config.cljs]: Done

File: src/status_im/subs/wallet/wallet.cljs (hunk 2 above)
Comment (mmilad75) [reply src/status_im/subs/wallet/wallet.cljs:296]: Done

**Comment by mmilad75** [comment]
> Hi @mmilad75 !
> 
> Checked and still see sum of My + Watch Only tokens: Screenshot 2024-04-25 at 19 51 08

Please check again with the latest build 🙏

**Review by smohamedjavid — APPROVED** [review]

**Comment by mariia-skrypnyk** [comment]
Thanks @mmilad75 !

Functionality works properly. 
Need to rerun tests and then let you know if all good.

**Review by vkjr — APPROVED** [review]

**Comment by mariia-skrypnyk** [comment]
Hi @mmilad75 !
Thanks for your PR, it is ready to merge.

**Pull request merged**